Tests/SwiftGenKitTests/StringsTests.swift

```diff
@@ -58,15 +58,109 @@ class StringsTests: XCTestCase {
 
   func testMultipleFilesDuplicate() throws {
     let parser = try Strings.Parser()
+    var receivedWarning: String?
+    parser.warningHandler = { message, _, _ -> Void in
+      receivedWarning = message
+    }
```

Comment:
Use an array instead to ensure we also catch cases where multiple warnings were emitted

```swift
var receivedWarnings: [String]
parser.warningHandler = { message, _, _ -> Void in
  receivedWarnings.append(message)
}
```

Then you can use `XCTAssertEqual(receivedWarnings. [expectedMessage])`. That will allow the test to also verify that we don't emit _others and unexpected_ warnings before that one.